jsx/Modal.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * This file contains the React Component for a Modal Window.
+ *
+ * @author Henri Rabalais
+ * @version 1.1.0
+ *
+ */
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import swal from 'sweetalert2';
+
+/**
+ * Modal Component.
+ * React wrapper for a Modal Window. Allows to dynamically toggle a Modal window.
+ *
+ * ================================================
+ * Usage:
+ * - Wrap the contents to be displayed by the Modal Window by the Modal Component.
+ * - Use the 'title' prop to set a title for the Modal Component.
+ * - Use the 'onSubmit' prop to set a submission promise object for the Modal's contents.
+ * - Use the 'throwWarning' prop to throw a warning upon closure of the Modal Window.
+ * =================================================
+ *
+ */
+class Modal extends Component {
+  constructor() {
+    super();
+    this.handleClose = this.handleClose.bind(this);
+  }
+
+  handleClose() {
+    if (this.props.throwWarning) {
+      swal({
+        title: 'Are You Sure?',
+        text: 'Leaving the form will result in the loss of any information' +
+          'entered.',
+        type: 'warning',
+        showCancelButton: true,
+        confirmButtonText: 'Proceed',
+        cancelButtonText: 'Cancel',
+      }).then((result) => result.value && this.props.close());
+    } else {
+      this.props.close();
+    }
+  }
+
+  render() {
+    const {open, children, onSubmit, title} = this.props;
```

Comment:
is open essentially show?

Reply:
Yeah. If it makes more sense, I can set the `state` in `TriggerableModal` to be called `open`, and the `props` with `Modal` to be called `show`.